### PR TITLE
[libbeat]: Add support for `values` in lowercase processor

### PR DESCRIPTION
**Pull request by khushijain21**
## Proposed commit message

This adds support to lowercase values. The user is expected to provide a full path (case sensitive here) of the key. With this addition, the user can now selectively convert the keys and values.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

Part of https://github.com/elastic/beats/issues/22254

### CODE REVIEW

File: libbeat/processors/actions/alterFieldProcessor.go [hunk 1]
```diff
@@ -133,3 +149,22 @@ func (a *alterFieldProcessor) alter(event *beat.Event, field string) error {
 
 	return nil
 }
+
+func (a *alterFieldProcessor) alterValue(event *beat.Event, value string) error {
```
Comment (rdner) [inline libbeat/processors/actions/alterFieldProcessor.go:153]: Better to rename, so it's clear this not the actual value.

```suggestion
func (a *alterFieldProcessor) alterValue(event *beat.Event, valueKey string) error {
```

This does not look like the implementation we have discussed before.

We need to:

1. Get a value for `valueKey`
2. Check if it's a string
3. If it's not a string – throw error
4. If it's a string replace it with the altered value.

For this I don't see how it's necessary to use `Traverse`, why not just `GetValue,Delete,Put`? If the end value is always a string and we don't change the `valueKey` it should be way simpler.

Is there something I'm missing that it has to use `Traverse`?

File: libbeat/processors/actions/docs/lowercase.asciidoc [hunk 2]
```diff
@@ -103,6 +107,7 @@ processors:
 The `lowercase` processor has the following configuration settings:
 
 `fields`:: The field names to lowercase. The match is case-insensitive, e.g. `a.b.c.d` would match `A.b.C.d` or `A.B.C.D`.
+`values`:: (Optional) Specifies the exact values to be converted to lowercase. Each entry should include the full path to the value. Key matching is case-sensitive.
```
Comment (rdner) [inline libbeat/processors/actions/docs/lowercase.asciidoc:110]:
```suggestion
`values`:: (Optional) Specifies the exact values to be converted to lowercase. Each entry should include the full path to the value. Key matching is case-sensitive. If the target value is not a string, an error is triggered (`fail_on_error: true`) or the value is skipped (`fail_on_error: false`).
```

File: libbeat/processors/actions/alterFieldProcessor.go (hunk 1 above)
Comment (khushijain21) [reply libbeat/processors/actions/alterFieldProcessor.go:153]: yes we can use `GetValue`, `Delete` and `Put`. Just both seemed okay

Comment (rdner) [reply libbeat/processors/actions/alterFieldProcessor.go:153]: I think when we don't need to modify fields and we match case-sensitively, we should just use `GetValue`, `Delete`, `Put`.

File: libbeat/processors/actions/alterFieldProcessor.go [hunk 3]
```diff
@@ -133,3 +149,27 @@ func (a *alterFieldProcessor) alter(event *beat.Event, field string) error {
...
+		return fmt.Errorf("could not fetch value for key: %s, Error: %w", valueKey, err)
+	}
+
+	if v, ok := value.(string); ok {
+		err = event.Delete(valueKey)
+		if err != nil {
+			return fmt.Errorf("could not delete key: %s,  %w", v, err)
+		}
+
+		v, _ = a.alterFunc(v)
```
Comment (rdner) [inline libbeat/processors/actions/alterFieldProcessor.go:165]: I think we should check errors here, and for the `alterField` function as well. 

For now it's just `strings.ToLower` but perhaps we would add more processors in the future and they would have an error case. We risk to miss these errors if this implementation is not updated accordingly.

Comment (khushijain21) [reply libbeat/processors/actions/alterFieldProcessor.go]: for `alterField` - and I think this piece of code handles it so should be good https://github.com/khushijain21/elastic-agent-libs/blob/main/mapstr/mapstr.go#L216

**Review by rdner — APPROVED** [review]
LGTM

**Review by mauri870 — COMMENTED** [review]
LGTM, thanks!

**Pull request merged**